features/integration/c2c.feature

```diff
@@ -0,0 +1,172 @@
+@c2c
+Feature: Contract to Contract Interaction
+  Scenario:
+    * an algod v2 client
+    * a kmd client
+    * wallet information
+    * suggested transaction parameters from the algod v2 client
+    * I create a new transient account and fund it with 100000000 microalgos.
+    * I make a transaction signer for the transient account.
+
+    ###### ------ app at context index 0: FakeRandom ------ ######
+    When I build an application transaction with the transient account, the current application, suggested params, operation "create", approval-program "programs/fake_random.teal", clear-program "programs/one.teal.tok", global-bytes 0, global-ints 1, local-bytes 0, local-ints 0, app-args "", foreign-apps "", foreign-assets "", app-accounts "", extra-pages 0
+    And I sign and submit the transaction, saving the txid. If there is an error it is "".
+    And I wait for the transaction to be confirmed.
+    Given I remember the new application ID.
+
+    Given a new AtomicTransactionComposer
+
+    # randInt(42) -> (r, witnessString)
+    Given I add the nonce "Thing One"
+    When I create the Method object from method signature "randInt(uint64)(uint64,byte[17])"
+    * I create a new method arguments array.
+    * I append the encoded arguments "AAAAAAAAACo=" to the method arguments array.
+    * I add a nonced method call with the transient account, the current application, suggested params, on complete "noop", current transaction signer, current method arguments.
+
+    # randInt(1337) -> (r, witnessString)
+    Given I add the nonce "Thing Two"
+    When I create the Method object from method signature "randInt(uint64)(uint64,byte[17])"
+    * I create a new method arguments array.
+    * I append the encoded arguments "AAAAAAAABTk=" to the method arguments array.
+    * I add a nonced method call with the transient account, the current application, suggested params, on complete "noop", current transaction signer, current method arguments.
+
+    Then I build the transaction group with the composer. If there is an error it is "".
+    Then I gather signatures with the composer.
+    And I execute the current transaction group with the composer.
+    Then The composer should have a status of "COMMITTED".
+    And The app should have returned ABI types "(uint64,byte[17]):(uint64,byte[17])".
+    And The 0th atomic result for randomInt(42) proves correct
```

Comment:
I think Cucumber interprets anything in parentheses in a special way (optional text), similar to curly braces: https://github.com/cucumber/cucumber-expressions#optional-text. Maybe we can wrap it around parentheses or wordify the argument instead of using naked parentheses.

Reply:
ouch... will investigate. But my first reaction is "really cucumber?"

Reply:
In JS, you just have to escape the parentheses within the step:
```
'The {int}th atomic result for randomInt\\({int}) proves correct'
```
It looks like the other SDKs do not suffer from the same fate so this comment can be resolved as is 😄